File: code/python/climbStairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        total = 0
        twos = 0
        num = n
        
        if n<=0:
            return 0
        
        if n<=3:
            return n
        
        
        while n >= 0:
            temp = 1
            for i in range(twos):
                temp *= (num-i)
            for i in range(twos):
                temp = temp//(twos-i)        
            total += temp   
            n = n - 2
            twos += 1
            num = n + twos
            
        return total

File: code/java/climbStairs.java
class Solution {
    public int climbStairs(int n) {
        int total = 0;
        int twos = 0;
        int num = n;
        
        if(n<=0)
            return 0;
        
        if(n<=3)
            return n;
        
        int[] nums = new int[n];
        nums[0] = 1;
        nums[1] = 2;
        for(int i=2;i<n;i++)
            nums[i] = nums[i-1] + nums[i-2];
        
        return nums[n-1];
    }
}
